beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -156,31 +161,53 @@ pub async fn fetch_and_process_engine_blobs<T: BeaconChainTypes>(
                         }
                     };
 
-                    // Check indices from cache before sending the columns, to make sure we don't
-                    // publish components already seen on gossip.
-                    let all_data_columns_iter = all_data_columns.clone().into_iter();
-                    let data_columns_to_publish = match chain_cloned
-                        .data_availability_checker
-                        .imported_custody_column_indexes(&block_root)
-                    {
-                        None => Either::Left(all_data_columns_iter),
-                        Some(imported_columns_indices) => Either::Right(
-                            all_data_columns_iter
-                                .filter(move |d| !imported_columns_indices.contains(&d.index())),
-                        ),
-                    }
-                    .collect::<Vec<_>>();
-
-                    if let Err(e) = data_columns_sender.try_send(all_data_columns) {
+                    if let Err(e) = data_columns_sender.try_send(all_data_columns.clone()) {
                         error!(logger, "Failed to send computed data columns"; "error" => ?e);
                     };
 
+                    // Check indices from cache before sending the columns, to make sure we don't
+                    // publish components already seen on gossip.
                     let is_supernode = chain_cloned
                         .data_availability_checker
                         .get_custody_columns_count()
                         == spec.number_of_columns;
-                    if is_supernode && !data_columns_to_publish.is_empty() {
-                        publish_fn(BlobsOrDataColumns::DataColumns(data_columns_to_publish));
+
+                    // At the moment non supernodes are not required to publish any columns.
+                    // TODO(das): we could experiment with having full nodes publish their custodied
+                    // columns here.
+                    if !is_supernode {
+                        return;
+                    }
+
+                    // To reduce bandwidth for supernodes: permute the columns to publish and
+                    // publish them in batches. Our hope is that some columns arrive from
+                    // other supernodes in the meantime, obviating the need for us to publish
+                    // them. If no other publisher exists for a column, it will eventually get
+                    // published here.
+                    // FIXME(das): deduplicate this wrt to gossip/sync methods
```

Comment:
Is this comment still relevant? looks like you're checking it below

Reply:
I mean deduplicating the code so that we're not repeating heaps of this logic

Reply:
Ah got it!